umbraco/Umbraco-CMS
# Media picker output expansion

### Prerequisites


### Description

Media pickers should not expand media custom properties by default, as this can yield cyclic references. Instead we need to:

1. Re-introduce the custom Umbraco properties (width, heigh, ...) on the output model.
2. Support query expansion like we do for content pickers.

### Testing this PR

1. Add custom fields to the "image" media type and fill them out on a media item.
   - include a media picker as a custom field and pick the media itself in this custom field (thus causing a cyclic reference).
2. Create content with a standard media picker (not the "image" media picker, you'll need to pick other media types later) and pick the media item.
3. Request this content through the Delivery API and verify that:
   - the media width, height, extension and bytes are part of the output model instead of in the properties.
   - the properties collection is empty by default.
   - the media property can be expanded by means of standard output expansion.
   - no cyclic reference errors occur when expanding the media property - the custom media picker field is not expanded.
 4. Pick a PDF file on the content item.
 5. Request the content through the Delivery API and verify that:
    - the width and height fields are null.
    - everything else looks as it should.
    - output expansion still works.
